pageserver/src/tenant/mgr.rs

```diff
@@ -206,19 +254,39 @@ pub async fn init_tenant_mgr(
                         }
                     };
 
+                    // Try loading the location configuration
+                    let location_conf = Tenant::load_tenant_config(conf, &tenant_id).ok();
+
                     let generation = if let Some(generations) = &tenant_generations {
                         // We have a generation map: treat it as the authority for whether
                         // this tenant is really attached.
                         if let Some(gen) = generations.get(&tenant_id) {
                             *gen
                         } else {
-                            info!("Detaching tenant {tenant_id}, control plane omitted it in re-attach response");
-                            if let Err(e) = safe_remove_tenant_dir_all(&tenant_dir_path).await {
-                                error!(
-                                    "Failed to remove detached tenant directory '{}': {:?}",
-                                    tenant_dir_path.display(),
-                                    e
-                                );
+                            let is_secondary = location_conf
+                                .map(|c| matches!(c.mode, LocationMode::Secondary(_)))
+                                .unwrap_or(false);
```

Comment:
Would prefer exhaustive match over `c.mode` for this entire `else` block.

Reply:
In e578ed589 I've done that, and also fixed how we load LocationConf to surface areas instead of using an Option here.